File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/base/BaseActivity.java
package ua.meugen.android.pdfcomposer.ui.activities.base;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import ua.meugen.android.pdfcomposer.app.PdfComposer;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable final Bundle savedInstanceState) {
        inject();
        super.onCreate(savedInstanceState);
    }

    private void inject() {
        final Injector injector = createInjector();
        injector.inject(PdfComposer.from(this).getAppComponent());
    }

    @NonNull
    protected abstract Injector createInjector();
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/base/BaseActivity.kt
package ua.meugen.android.pdfcomposer.ui.activities.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import ua.meugen.android.pdfcomposer.app.PdfComposer

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        inject()
        super.onCreate(savedInstanceState)
    }

    private fun inject() {
        val injector = createInjector()
        injector.inject(PdfComposer.from(this).appComponent)
    }

    protected abstract fun createInjector(): Injector
}
